WordPress/WordPressTest/StoreCoordinatorTests.swift

```diff
@@ -56,32 +62,74 @@ class StoreCoordinatorTests: XCTestCase {
         XCTAssertEqual(availability, PurchaseAvailability.available)
     }
 
+    func testCannotPurchaseWhenPurchaseAlreadyPending() {
+        let payment: PendingPayment = (premium.id, testProduct, testSite)
+        let coordinator = storeCoordinator(paymentsEnabled: true, pending: payment)
+        let product = productForPlan(TestPlans.business.plan)
+
+        // And now attempt a second purchase
+        XCTAssertThrowsError(try coordinator.purchasePlan(business, product: product, forSite: testSite))
+    }
+    
+    func testCanMakePaymentWhenNoPaymentIsPending() {
+        let coordinator = storeCoordinator(paymentsEnabled: true, pending: nil)
+        let product = productForPlan(TestPlans.business.plan)
+
+        do {
+            try coordinator.purchasePlan(business, product: product, forSite: otherSite)
+        } catch {
+            XCTFail("Expected call not to throw")
+        }
+    }
+    
     // ========================= END OF TESTS =============================== //
 
 
     // MARK: - Helpers
 
     private func availabilityWith(plan plan: Plan, active: Plan, paymentsEnabled: Bool, pendingState: PendingState) -> PurchaseAvailability {
-        let pendingPayment = pending(plan: plan, siteID: testSite, state: pendingState)
+        let pendingPayment = pending(plan: plan, productID: testProduct, siteID: testSite, state: pendingState)
         let coordinator = storeCoordinator(paymentsEnabled: paymentsEnabled, pending: pendingPayment)
         return coordinator.purchaseAvailability(forPlan: plan, siteID: testSite, activePlan: active)
     }
 
-    private func storeCoordinator(paymentsEnabled paymentsEnabled: Bool, pending: (Plan, Int)?) -> StoreCoordinator<MockStore> {
+    private func storeCoordinator(paymentsEnabled paymentsEnabled: Bool, pending: PendingPayment?) -> StoreCoordinator<MockStore> {
         var store = MockStore.succeeding()
         store.canMakePayments = paymentsEnabled
+ 
         let coordinator = StoreCoordinator(store: store)
-        coordinator.pendingPayment = pending
+        
+        if let pending = pending,
+            let plan = TestPlans.allPlans.filter({ $0.id == pending.planID }).first {
+            let product = productForPlan(plan)
+            try! coordinator.purchasePlan(plan, product: product, forSite: pending.siteID)
+        }
+        
         return coordinator
     }
+    
+    private func clearUserDefaults() {
+        let defaults = NSUserDefaults.standardUserDefaults()
+        defaults.removeObjectForKey("PendingPaymentPlanIDUserDefaultsKey")
+        defaults.removeObjectForKey("PendingPaymentProductIDUserDefaultsKey")
+        defaults.removeObjectForKey("PendingPaymentSiteIDUserDefaultsKey")
```

Comment:
Good idea. I figured the options were to suck it up and clear the user defaults here (I agree I don't like replicating these implementation details in the tests) or inject some storage into StoreCoordinator. We can tackle this in #5316.